mbrl/util/replay_buffer.py

```diff
@@ -163,6 +163,99 @@ def ensemble_size(self):
         return self._ensemble_size
 
 
+class SequenceTransitionIterator(BootstrapIterator):
+    """
+    Provides an iterator which is compatible with BootstrapIterator. Batches consist of short
+    sequences from the trajectory which can overlap. The correlation in the batches can be
+    reduced by a smaller sequence length. Note that the true batch size corresponds to the
+    specified one times the sequence length.
+    """
+
+    def __init__(
+        self,
+        transitions: TransitionBatch,
+        trajectory_indices: List[Tuple[int, int]],
+        batch_size: int,
+        sequence_length: int,
+        ensemble_size: int,
+        shuffle_each_epoch: bool = False,
+        use_last_transition=True,
+        rng: Optional[np.random.Generator] = None,
+    ):
+        self._sequence_length = sequence_length
+        # filter trajectories of length shorter than sequence_length
+        self._truncate_by = 1 if use_last_transition else 2
+        self._trajectory_indices = list(
+            filter(
+                lambda tr: (tr[1] - tr[0]) >= sequence_length + self._truncate_by - 1,
+                trajectory_indices,
+            )
+        )
+        if len(self._trajectory_indices) < 0.5 * len(trajectory_indices):
+            warnings.warn(
+                "More than 50% of trajectories are smaller than the specified length."
+            )
+        self._sequence_length = sequence_length
+        self._bootstrap_iter = True
+        self._use_last_transition = use_last_transition
+        super().__init__(
+            transitions,
+            batch_size * sequence_length,  # this is a hack to stay compatible.
+            ensemble_size,
+            shuffle_each_epoch,
+            False,
+            rng,
+        )
+
+    def _sample_member_indices(self) -> np.ndarray:
+        self.member_indices = np.empty((self.ensemble_size, self.batch_size), dtype=int)
+        truncated_trajectory_indices = np.array(self._trajectory_indices)
+        truncated_trajectory_indices[:, 1] -= self._sequence_length - self._truncate_by
```

Comment:
I'm also a bit unclear on the purpose of `self._truncate_by` (even with the default value of 1)

Reply:
You are right. This actually only works in the default case.
The correct version would be
```suggestion
        self._truncate_by = 0 if use_last_transition else 1
        ...
        truncated_trajectory_indices[:, 1] -= self._sequence_length - 1 + self._truncate_by
```
So yes, the idea is simply to truncate the start positions, since I assumed the trajectories were given in the form of [start_position, end_position).